scripts/tasks/fluentui-publish.ts

```diff
@@ -1,33 +1,163 @@
-      argv().registry,
-      bumpType,
-    ];
-
-    logger.info(`Running this command: yarn ${lernaPublishArgs.join(' ')}`);
-
-    const result = spawnSync('yarn', lernaPublishArgs, {
-      cwd: fluentRoot,
-      shell: true,
-      stdio: 'inherit',
-    });
+    const gitRoot = findGitRoot();
 
-    if (result.status) {
-      throw new Error(result.error?.stack || `lerna publish failed with status ${result.status}`);
+    const branch = execCommandSync(gitRoot, 'git', ['branch', '--show-current']);
+    execCommandSync(gitRoot, 'git', ['fetch', 'origin']);
+    execCommandSync(gitRoot, 'git', ['reset', '--hard', `origin/${branch}`]); // sometimes lerna add gitHead in package.json after release
+
+    // sync fluent version
+    execCommandSync(gitRoot, 'yarn', ['syncpack:fix']);
+    const gitStatus = execCommandSync(gitRoot, 'git', ['status']);
+    if (!gitStatus.includes('nothing to commit, working tree clean')) {
+      execCommandSync(gitRoot, 'git', ['add', `\*package.json`]);
+      execCommandSync(gitRoot, 'git', ['commit', '-m', `"chore: fix dependencies after react-northstar release"`]);
+      execCommandSync(gitRoot, 'git', ['push']);
     }
+
+    // make sure there's no more than one fluent versions
+    execCommandSync(gitRoot, 'yarn', ['syncpack:list']);
+  };
+}
+
+// pack all public fluent ui packages, used by ci to store nightly built artifacts
+const PUBLIC_FLUENT_PACKAGES = [
+  'ability-attributes',
+  'accessibility',
+  'code-sandbox',
+  'docs-components',
+  'react-bindings',
+  'react-builder',
+  'react-component-event-listener',
+  'react-component-nesting-registry',
+  'react-component-ref',
+  'react-context-selector',
+  'react-icons-northstar',
+  'react-northstar-emotion-renderer',
+  'react-northstar-fela-renderer',
+  'react-northstar-styles-renderer',
+  'react-northstar',
+  'react-proptypes',
+  'react-telemetry',
+  'state',
+  'styles',
+];
+
+const TODAY = new Date().toISOString().split('T')[0]; // yyyy-mm-dd
+
+const replaceDepVersionWithNightlyUrl = (packageRoot, packageName) => {
+  const packageJson = require(`@fluentui/${packageName}/package.json`);
+  const dependencies = packageJson.dependencies || {};
+  let hasFluentDependency = false;
+  for (const key of Object.keys(dependencies)) {
+    if (key.startsWith('@fluentui/') && PUBLIC_FLUENT_PACKAGES.includes(fluentPackageName)) {
+      const fluentPackageName = key.substring('@fluentui/'.length);
+      hasFluentDependency = true;
+      dependencies[
+        key
+      ] = `https://fluentsite.blob.core.windows.net/nightly-builds/${TODAY}/fluentui-${fluentPackageName}-${packageJson.version}.tgz`;
+    }
+  }
+
+  if (hasFluentDependency) {
+    fs.writeFileSync(path.resolve(packageRoot, 'package.json'), JSON.stringify(packageJson, null, 2));
+  }
+};
+
+export function packFluentTarballs() {
+  return function() {
+    const gitRoot = findGitRoot();
+    const tempFolderForPacks = path.resolve(gitRoot, 'fluentui-nightly');
```

Comment:
can we make this path a cli arg or env variable passed from the pipeline ? right now there's two places this path is declared and if one is changed with out the other

💣 💣 💣

Reply:
and if somehow the task in the pipeline doesn't actually throw when it can't find the path, it will be even more interesting 1 month later when we start asking where the nightly builds are

Reply:
Good point, let me make this a variable